epinar/assignment5/unknown_alphabet.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+class GraphNode:
+
+	def __init__(self, v):
+		self.value = v
+		self.indegree = 0
+		self.edges = set()
+		self.is_visited = False
+
+	def add_edge(self, w):
+		self.edges.add(w)
+		w.indegree += 1
+
+
+class WordGraph:
+
+	def __init__(self, dictionary):
+		letters = set()
+		for word in dictionary:
+			letters = letters.union(word)
+
+		self.nodes = {alpha: GraphNode(alpha) for alpha in letters}
+
+		for word1, word2 in zip(dictionary, dictionary[1:]):
+			cp = _common_prefix(word1, word2)
+			if cp != -1:
+				self.nodes[word1[cp]].add_edge(self.nodes[word2[cp]])
+
+	def __iter__(self):
+		for node in self.nodes.values():
+			yield node
+
+
+def _topsort1(v, graph, alphabet):
```

Comment:
I'm not sure that we need this function if we have a "_topsort2"

Reply:
removed 👍